bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,256 @@
+            cursor.kind.name,
+            cursor.spelling,
+        )
 
+    # Get cursor's children and recursively print
+    for child_node in valid_children(node):
+        print_ast(child_node)
+
+
+def generate_parsed_info(node):
+    """
+    Generates parsed information by recursively traversing the AST
+
+    Arguments:
+        - node (dict):
+            - The node in the AST
+            - Keys:
+                - cursor: The cursor pointing to a node
+                - filename: 
+                    - The file's name to check if the node belongs to it
+                    - Needed to ensure that only symbols belonging to the file gets parsed, not the included files' symbols
+                - depth: The depth of the node (root=0)
+
+    Returns:
+        - parsed_info (dict): 
+            - Contains key-value pairs of various traits of a node
+            - The key 'members' contains the node's children's `parsed_info`
+    """
+
+    parsed_info = dict()
+
+    cursor = node["cursor"]
+    depth = node["depth"]
+
+    # @TODO: fix
+    if cursor.spelling != "":
+        parsed_info["depth"] = depth
+        parsed_info["line"] = cursor.location.line
+        parsed_info["column"] = cursor.location.column
+        parsed_info["kind"] = cursor.kind.name
+        parsed_info["name"] = cursor.spelling
+        if cursor.type.kind.spelling != "Invalid":
+            parsed_info["element_type"] = cursor.type.kind.spelling
+        if cursor.access_specifier.name != "INVALID":
+            parsed_info["access_specifier"] = cursor.access_specifier.name
+        if cursor.result_type.spelling != "":
+            parsed_info["result_type"] = cursor.result_type.spelling
+        if cursor.brief_comment:
+            parsed_info["brief_comment"] = cursor.brief_comment
+        if cursor.raw_comment:
+            parsed_info["raw_comment"] = cursor.raw_comment
+        parsed_info["members"] = []
 
-def print_node(cursor, lines, depth):
-    file = cursor.location.file
-    line, column = cursor.location.line, cursor.location.column
-    result_type = (
-        cursor.result_type.spelling if cursor.result_type.spelling is not "" else "None"
-    )
-    print(
-        "-" * depth,
-        file,
-        f"L{line} C{column}",
-        bcolors.BOLD + cursor.kind.name + bcolors.ENDC,
-        bcolors.OKBLUE + cursor.spelling + bcolors.ENDC,
-        bcolors.OKGREEN + cursor.access_specifier.name + bcolors.ENDC,
-        bcolors.OKGREEN + "->" + bcolors.ENDC,
-        bcolors.OKGREEN + result_type + bcolors.ENDC,
-    )
+    # Get cursor's children and recursively add their info to a dictionary, as members of the parent
+    for child_node in valid_children(node):
+        child_parsed_info = generate_parsed_info(child_node)
+        # If both child and parent's info is not empty (opening check for dictionary population), add child's info to parent's members
+        if child_parsed_info and parsed_info:
```

Comment:
This is more pythonic than testing for `len(x) != 0`, well done

Reply:
Thank you!